core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -137,28 +137,78 @@ def __init__(self,
-        self._user_params = hyperparameters  # TODO: Delete after initialization?
-        if self._user_params is None:
-            self._user_params = dict()
-
         self.params_trained = dict()
         self._is_initialized = False
         self._is_fit_metadata_registered = False
         self._fit_metadata = dict()
 
         self._compiler = None
 
+    @classmethod
+    def _init_user_params(cls, params: Optional[Dict[str, Any]] = None, ag_args_fit: str = AG_ARGS_FIT, ag_arg_prefix: str = AG_ARG_PREFIX) -> (Dict[str, Any], Dict[str, Any]):
+        """
+        Given the user-specified hyperparameters, split into `params` and `params_aux`.
+
+        Parameters
+        ----------
+        params : Optional[Dict[str, Any]], default = None
+            The model hyperparameters dictionary
+        ag_args_fit : str, default = "ag_args_fit"
+            The params key to look for that contains params_aux.
+            If the key is present, the value is used for params_aux and popped from params.
+            If no such key is found, then initialize params_aux as an empty dictionary.
+        ag_arg_prefix : str, default = "ag."
+            The key prefix to look for that indicates a parameter is intended for params_aux.
+            If None, this logic is skipped.
+            If a key starts with this prefix, it is popped from params and added to params_aux with the prefix removed.
+            For example:
+                input:  params={'ag.foo': 2, 'abc': 7}, params_aux={'bar': 3}, and ag_arg_prefix='.ag',
+                output: params={'abc': 7}, params_aux={'bar': 3, 'foo': 2}
+            In cases where the key is specified multiple times, the value of the key with the prefix will always take priority.
+            A warning will be logged if a key is present multiple times.
+            For example, given the most complex scenario:
+                input:  params={'ag.foo': 1, 'foo': 2, 'ag_args_fit': {'ag.foo': 3, 'foo': 4}}
+                output: params={'foo': 2}, params_aux={'foo': 1}
+
+        Returns
+        -------
+        params, params_aux : (Dict[str, Any], Dict[str, Any])
+            params will contain the native model hyperparameters
+            params_aux will contain special auxiliary hyperparameters
+        """
+        params = copy.deepcopy(params) if params is not None else dict()
+        assert isinstance(params, dict), f"Invalid dtype of params! Expected dict, but got {type(params)}"
+        params_aux = params.pop(ag_args_fit, dict())
+        if params_aux is None:
+            params_aux = dict()
+        assert isinstance(params_aux, dict), f"Invalid dtype of params_aux! Expected dict, but got {type(params_aux)}"
+        if ag_arg_prefix is not None:
+            param_aux_keys = list(params_aux.keys())
+            for k in param_aux_keys:
+                if isinstance(k, str) and k.startswith(ag_arg_prefix):
+                    k_no_prefix = k[len(ag_arg_prefix):]
```

Comment:
do we expect just one level of nesting? Would it bring value supporting multi-level nesting?

Reply:
We could add in future, but there is nothing implemented that would take advantage of it currently.